src/sentry/sentry_metrics/aggregation_option_registry.py

```diff
@@ -22,17 +23,41 @@ class TimeWindow(Enum):
     "d:transactions/measurements.lcp@millisecond": {AggregationOption.HIST: TimeWindow.NINETY_DAYS},
 }
 
-USE_CASE_AGG_OPTION = {UseCaseID.CUSTOM: {AggregationOption.TEN_SECOND: TimeWindow.SEVEN_DAYS}}
+# Currently there are no default per-use case aggregation overrides
+# They are all set via options
+USE_CASE_AGG_OPTION = {}
 
 
-def get_aggregation_options(mri: str) -> dict[AggregationOption, TimeWindow] | None:
+def set_use_case_aggregation_options():
+    # As of now this option is still restricted
+    # to the scope of the custom use case
+    if options.get("sentry-metrics.10s-granularity"):
+        USE_CASE_AGG_OPTION[UseCaseID.CUSTOM] = {
+            AggregationOption.TEN_SECOND: TimeWindow.SEVEN_DAYS
+        }
+
+    for use_case in options.get("sentry-metrics.drop-percentiles.per-use-case"):
+        USE_CASE_AGG_OPTION[UseCaseID(use_case)] = {
+            AggregationOption.NO_PERCENTILE: TimeWindow.NINETY_DAYS
+        }
+
+
+def get_aggregation_options(mri: str, org_id: int) -> dict[AggregationOption, TimeWindow] | None:
```

Comment:
Since we are only able to write 1 aggregation option at the moment, what if we added a test case which validated that assumption. I am not exactly sure whether that constraint is enforced somewhere in the code or not. But it would be good to have the constrain in place to avoid potential problems in the near future.

Reply:
Good point. Added some logic to our existing unit tests where there are a variety of aggregation options set up, which cover all the use cases etc. The unit test asserts that only 1 exists per metric bucket payload, though. 

Once we have multiple aggregation option support, we can remove and refactor these tests to communicate that change in expectations.

Reply:
The indexer already assumes a `dict` output and basically chooses the first element. So in any case, we are always guaranteed to have 1 aggregation option per payload. 

I can change the typing and indexer behavior in a subsequent PR, but for now I think I'd like to not have this PR touch too many files at once.